### PR TITLE
Update bilibili extension

### PR DESCRIPTION
**Pull request by Fatpandac**
## Description
Implement filter

## Screencast
image

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/Bilibili/src/utils/api.ts [hunk 1]
```diff
@@ -1,13 +1,18 @@
+const BASE_PASSPORT_URL = "http://passport.bilibili.com";
```
Comment (pernielsentikaer) [inline extensions/Bilibili/src/utils/api.ts:1]: https?

File: extensions/Bilibili/src/dynamicFeed.tsx [hunk 2]
```diff
@@ -1,24 +1,66 @@
...
+  const [filterType, setFilterType] = useState("");
   const { dynamicItems, isLoading } = useDynamicFeed();
 
   const videoTypes = ["DYNAMIC_TYPE_LIVE_RCMD", "DYNAMIC_TYPE_AV"];
 
+  const kindTypes: KindType[] = [
+    { id: "0", name: "全部" },
+    { id: "1", name: "未看过的视频" },
+    { id: "2", name: "看过的视频" },
+    { id: "3", name: "视频" },
```
Comment (pernielsentikaer) [inline extensions/Bilibili/src/dynamicFeed.tsx:40]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 🙂 

- Could you also add a [CHANGELOG](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history) entry?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/Bilibili/src/utils/api.ts (hunk 1 above)
Comment (Fatpandac) [reply extensions/Bilibili/src/utils/api.ts:1]: Oh, this API supports both HTTP and HTTPS. I’ll change it later.

File: extensions/Bilibili/src/dynamicFeed.tsx (hunk 2 above)
Comment (Fatpandac) [reply extensions/Bilibili/src/dynamicFeed.tsx:40]: Okay

File: extensions/Bilibili/src/dynamicFeed.tsx [hunk 3]
```diff
@@ -1,24 +1,66 @@
...
 
-interface Preferences {
-  justShowVideos: boolean;
+type KindType = { id: string; name: string };
+
+function DrinkDropdown(props: { kindTypes: KindType[]; onKindTypeChange: (newValue: string) => void }) {
+  const { kindTypes, onKindTypeChange } = props;
+  return (
+    <List.Dropdown
+      tooltip="Select Drink Type"
```
Comment (pernielsentikaer) [inline extensions/Bilibili/src/dynamicFeed.tsx:13]: Could you correct this

File: extensions/Bilibili/src/dynamicFeed.tsx [hunk 4]
```diff
@@ -1,24 +1,66 @@
...
+      storeValue={true}
+      onChange={(newValue) => {
+        onKindTypeChange(newValue);
+      }}
+    >
+      <List.Dropdown.Section title="Alcoholic Beverages">
```
Comment (pernielsentikaer) [inline extensions/Bilibili/src/dynamicFeed.tsx:19]: Could you correct this

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Just saw two small NITs, could you correct them?

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**